oio/xcute/orchestrator.py

```diff
@@ -116,8 +136,15 @@ def orchestrate_loop(self):
             One iteration of the main loop
         """
 
-        new_jobs = iter(
-            lambda: self.backend.run_next(self.orchestrator_id), None)
+        def run_next_job():
+            next_job, exc = self.handle_backend_errors(
+                self.backend.run_next, self.orchestrator_id)
+            if exc is None:
+                return next_job
+            self.logger.warn('Unable to run next job: %s', exc)
+            return None
+
+        new_jobs = iter(lambda: run_next_job(), None)
```

Comment:
If the code has become this complicated, I don't think there's any point in using `iter(iterable, None)` anymore, is it?